src/util.ts

```diff
@@ -40,3 +51,54 @@ export function safeStatSync(fn: string): fs.Stats {
     return null;
   }
 }
+
+/**
+ * Like Promise.all, but also displays a progress bar that fills as the
+ * promises resolve. The label is a helpful string describing the operation
+ * that the user is waiting on.
+ */
+export function promiseAllWithProgress<T>(
+    promises: Promise<T>[], label: string): Promise<T[]> {
+  if (promises.length === 0) {
+    return Promise.all([]);
+  }
+  const progressBar = standardProgressBar(label, promises.length);
+  const progressed: Promise<T>[] = [];
+  for (const promise of promises) {
```

Comment:
I think a .map would work well here, especially because with an async mapper you can use `await` and it'll also tick on errors (maybe, maybe not better):

```typescript
return Promise.all(promises.map(async (p) => {
  const result = await p;
  progressBar.tick();
  return result;
}));
```